### PR TITLE
New serverless pattern - apigw-lambda-translate-cdk

### PR DESCRIPTION
**Pull request by awsarippa**
*Issue #, if available:*

*Description of changes:*
This patterns shows CDK deployment on how to leverage Amazon API Gateway, AWS Lambda, and Amazon Translate to perform language translation in a serverless fashion.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apigw-lambda-translate-cdk/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+```
+and deploy with
+```
+cdk deploy
+```
+
+The deployment will create a S3 bucket, a Lambda function and a DynamoDB table.
+
+## How it works
+The API Gateway handles the incoming requests from user and it invokes the relevant route. The Lambda, triggered by API Gateway, invokes the Translate's TranslateText API and the analyzed response from Translate service is routed back to the requester.
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/README.md:68]:
```suggestion
The API Gateway handles the incoming requests from the user and invokes the relevant route. The Lambda function, triggered by API Gateway, invokes Translate's TranslateText API and the analyzed response from Translate is routed back to the requester.
```

File: apigw-lambda-translate-cdk/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+All AWS CDK v2 deployments use dedicated AWS resources to hold data during deployment. Therefore, your AWS account and Region must be bootstrapped to create these resources before you can deploy. If you haven't already bootstrapped execute the below command
+```
+cdk bootstrap
+```
+and deploy with
+```
+cdk deploy
+```
+
+The deployment will create a S3 bucket, a Lambda function and a DynamoDB table.
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/README.md:65]: Did you forget to update these to API GW etc.?

File: apigw-lambda-translate-cdk/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## How it works
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/README.md:67]: Consider adding a statement that the text is translated to French by default and how the user can change it if needed

File: apigw-lambda-translate-cdk/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# API Gateway -> Lambda -> Translate
+This patterns shows CDK deployment on how to leverage Amazon API Gateway, AWS Lambda, and Amazon Translate to perform language translation in a serverless fashion.
+
+## Architecture
+![Diagram](src/architecture.png)
+
+### What resources will be created?
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/README.md:7]: Please add a line that the API GW is not using authorization or authentication and should only be used for demo purposes

File: apigw-lambda-translate-cdk/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    },
+    "gitHub": {
+      "template": {
+        "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/apigw-lambda-translate-cdk",
+        "templateURL": "serverless-patterns/apigw-lambda-translate-cdk",
+        "projectFolder": "apigw-lambda-translate-cdk",
+        "templateFile": "app.py"
+      }
+    },
+    "resources": {
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/example-pattern.json:21]: please add a link to API GW documentation as well

File: apigw-lambda-translate-cdk/src/architecture.png
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/src/architecture.png:1]: Please use up to date icons from https://aws.amazon.com/architecture/icons/

File: apigw-lambda-translate-cdk/README.md (hunk 4 above)
Comment (awsarippa) [reply apigw-lambda-translate-cdk/README.md:7]: This is done. Updated README.md with caution note.

File: apigw-lambda-translate-cdk/README.md (hunk 3 above)
Comment (awsarippa) [reply apigw-lambda-translate-cdk/README.md:67]: This is done. Added note saying the default language for translation is set to French and that users are requested to change it as per their use-case.

File: apigw-lambda-translate-cdk/example-pattern.json (hunk 5 above)
Comment (awsarippa) [reply apigw-lambda-translate-cdk/example-pattern.json:21]: Done. Added documentation links for developer guide and features of API Gateway.

File: apigw-lambda-translate-cdk/README.md (hunk 2 above)
Comment (awsarippa) [reply apigw-lambda-translate-cdk/README.md]: Added the API Gateway info. Missed it.

File: apigw-lambda-translate-cdk/src/architecture.png
Comment (awsarippa) [reply apigw-lambda-translate-cdk/src/architecture.png:1]: Done. Updated the archie diagram.

File: apigw-lambda-translate-cdk/README.md (hunk 1 above)
Comment (awsarippa) [reply apigw-lambda-translate-cdk/README.md]: Done. Updated the comment.

**Comment by awsarippa** [comment]
@bfreiberg Thanks a lot for your comments and review. I've committed the changes as suggested by you.

File: apigw-lambda-translate-cdk/apigw-lambda-translate-cdk.json [hunk 6]
```diff
@@ -0,0 +1,94 @@
+{
+    "title": "API Gateway - Lambda - Translate",
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/apigw-lambda-translate-cdk.json:2]:
```suggestion
    "title": "API Gateway to Lambda to Translate",
```

File: apigw-lambda-translate-cdk/apigw-lambda-translate-cdk.json [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    "description": "Perform language translation using Translate through Lambda and API Gateway",
+    "language": "Python",
+    "level": "200",
+    "framework": "CDK",
+    "introBox": {
+        "headline": "How it works",
+        "text": [
+            "This patterns shows CDK deployment on how to leverage Amazon API Gateway, AWS Lambda, and Amazon Translate to perform language translation in a serverless fashion."
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/apigw-lambda-translate-cdk.json:10]:
```suggestion
            "How to leverage Amazon API Gateway, AWS Lambda, and Amazon Translate to perform language translation in a serverless fashion."
```

File: apigw-lambda-translate-cdk/apigw-lambda-translate-cdk.json [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        ]
+    },
+    "testing": {
+        "text": [
+            "See the GitHub repo for detailed testing instructions."
+        ]
+    },
+    "cleanup": {
+        "text": [
+            "Delete the stack: <code>cdk destroy</code>."
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/apigw-lambda-translate-cdk.json:53]:
```suggestion
            "<code>cdk destroy</code>"
```

File: apigw-lambda-translate-cdk/apigw-lambda-translate-cdk.json [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+            },
+            {
+                "text": "Amazon API Gateway Features",
+                "link": "https://aws.amazon.com/api-gateway/"
+            }
+        ]
+    },
+    "deploy": {
+        "text": [
+            "cdk deploy"
```
Comment (bfreiberg) [inline apigw-lambda-translate-cdk/apigw-lambda-translate-cdk.json:43]:
```suggestion
            "<code>cdk deploy</code>"
```

**Comment by awsarippa** [comment]
@bfreiberg - Completed and committed all suggestions.

**Comment by bfreiberg** [comment]
Looks good, thanks for your contribution.

**Comment by awsarippa** [comment]
Thnx @bfreiberg for your review and suggestions.

**Comment by bls20AWS** [comment]
live: https://serverlessland.com/patterns/apigw-lambda-translate-cdk

**Comment by awsarippa** [comment]
@bls20AWS - Thanks a lot Ben.

**Pull request merged**